docker/indexer/stages/processing/processing.go

```diff
@@ -0,0 +1,132 @@
+// Package processing implements the hashing step for each provide input.
+package processing
+
+import (
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/go-git/go-git/v5"
+	"github.com/google/osv.dev/docker/indexer/shared"
+	"github.com/google/osv.dev/docker/indexer/stages/preparation"
+
+	log "github.com/golang/glog"
+	pb "github.com/google/osv.dev/docker/indexer/proto"
+)
+
+const workers = 25
+
+// Storer is used to permanently store the results.
+type Storer interface {
+	Store(ctx context.Context, repoInfo *preparation.Result, fileResults []FileResult) error
+}
+
+// FileResult holds the per file hash and path information.
+type FileResult struct {
+	Path string
+	Hash [md5.Size]byte
```

Comment:
Maybe we should support multiple hashes here. 

E.g. if we want to extend this to support multiple hashing methods (e.g. one where we "normalize" the input file by formatting/removing comments), we can just add it here without changing any underlying data structures.

Reply:
changed `Hash` to a slice and added a field to define the hash type